apps/wallets/smart-wallet/react/src/app/env.ts

```diff
@@ -0,0 +1,37 @@
+import { z } from "zod";
+
+const envSchema = z.object({
+    REACT_APP_CROSSMINT_API_KEY_PROD: z.string(),
+    REACT_APP_CROSSMINT_API_KEY_STG: z.string().min(1),
```

Comment:
Or maybe we can specify somehow in zod that we need at least oen of these two, but dont need both?

Reply:
One way to achieve this would be something like:

```ts
const clientSchema = z
    .object({
        REACT_APP_CROSSMINT_API_KEY_PROD: z.string().optional(),
        REACT_APP_CROSSMINT_API_KEY_STG: z.string().optional(),
        REACT_APP_FIREBASE_API_KEY: z.string().min(1),
        REACT_APP_FIREBASE_AUTH_DOMAIN: z.string().min(1),
        REACT_APP_FIREBASE_PROJECT_ID: z.string().min(1),
        REACT_APP_FIREBASE_STORAGE_BUCKET: z.string().min(1),
        REACT_APP_FIREBASE_MESSAGING_SENDER_ID: z.string().min(1),
        REACT_APP_FIREBASE_APP_ID: z.string().min(1),
        REACT_APP_WEB3_AUTH_CLIENT_ID_PROD: z.string().optional(),
        REACT_APP_WEB3_AUTH_CLIENT_ID_STG: z.string().optional(),
        REACT_APP_WEB3_AUTH_NETWORK_PROD: z.string().optional(),
        REACT_APP_WEB3_AUTH_NETWORK_STG: z.string().optional(),
        REACT_APP_WEB3_AUTH_VERIFIER_ID_PROD: z.string().optional(),
        REACT_APP_WEB3_AUTH_VERIFIER_ID_STG: z.string().optional(),
    })
    .refine(
        (data) => {
            // Custom validation logic
            const keys = Object.keys(data);
            for (let i = 0; i < keys.length; i += 2) {
                const prodKey = keys[i];
                const stgKey = keys[i + 1];
                // Check if both or neither are defined
                if (
                    ((data as any)[prodKey] && (data as any)[stgKey]) ||
                    (!(data as any)[prodKey] && !(data as any)[stgKey])
                ) {
                    return false;
                }
            }
            return true;
        },
        {
            message: "For each prod/stg pair, exactly one must be defined.",
        }
    );

...
```

but i notice we run into type inference issues.

Reply:
@jmderby I would just do whatever is easiest here for now.

The demo improvement project next cycle will include dropping that chain toggle on the home page, meaning we can have a single set of env vars.